### PR TITLE
feat(idpe-17887): enable `/` in db name for v1 write.

### PR DESCRIPTION
**Pull request by wiedld**
Part of https://github.com/influxdata/idpe/issues/17887

~Enable v1 `/write` endpoint to support a db with `/`, if and only if, there is no named rp.~ Always concat db/rp provided to the write v1 endpoint, if not default nor autogen.

See test cases and error docs (in first commit) for the contract update.

## Checklist:
- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: router/src/server/http/write/single_tenant/mod.rs [hunk 1]
```diff
@@ -316,24 +313,44 @@ mod tests {
...
+    // `/` character is not permitted in the DB, if a named RP is specified
+    test_parse_v1!(
+        db_with_rp_separator_and_rp,
+        query_string = "?db=foo/bar&rp=bar",
+        want = Err(Error::SingleTenantError(
+            SingleTenantExtractError::ParseV1Request(V1WriteParseError::ContainsMultipleSeparators)
+        ))
+    );
+
+    // `/` character is allowed in the DB, if an autogen RP is specified
```
Comment (stuartcarnie) [inline router/src/server/http/write/single_tenant/mod.rs:344]: `default` is also a valid retention policy, and is treated the same way as `autogen` from the `/query` perspective.

**Review by stuartcarnie — COMMENTED** [review]
I think this looks good, @wiedld. I added a note to add `default` as a special retention policy, as it is treated the same as `autogen`:

https://github.com/influxdata/idpe/blob/837a193964f591dc6a76bb8cc03f54b070c50377/influxqlbridge/internal/server/statement.go#L48-L50

File: router/src/server/http/write/single_tenant/mod.rs [hunk 2]
```diff
@@ -316,24 +313,54 @@ mod tests {
...
     test_parse_v1!(
         rp_with_rp_separator,
         query_string = "?db=bananas&rp=are/great",
         want = Err(Error::SingleTenantError(
-            SingleTenantExtractError::ParseV1Request(V1WriteParseError::ContainsRpSeparator)
+            SingleTenantExtractError::ParseV1Request(V1WriteParseError::ContainsSeparator)
         ))
     );
 
+    // `/` character is not permitted in the DB, if a named RP is specified
```
[the remark this answers is not in the document]
Comment (wiedld) [reply router/src/server/http/write/single_tenant/mod.rs:335]: Had conversation within [internal slack](https://influxdata.slack.com/archives/C04K1QRSDFG/p1689622337922069?thread_ts=1689246690.462999&cid=C04K1QRSDFG). We are going to always concat the db and rp, even if that results in a bad namespace lookup with `/write?db=db/rp&rp=rp` => namespace `db/rp/rp`.

This behavior matches the query endpoint.

File: router/src/server/http/write/single_tenant/mod.rs (hunk 1 above)
Comment (wiedld) [reply router/src/server/http/write/single_tenant/mod.rs:344]: Ty. Updated.

**Review by mhilton — APPROVED** [review]

**Pull request merged**